staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -282,6 +282,8 @@ const (
 	NamespaceSystem string = "kube-system"
 	// NamespacePublic is the namespace where we place public info (ConfigMaps)
 	NamespacePublic string = "kube-public"
+	// NamespaceNodeLease is the namespace where we place node lease objects (used for node heartbeats)
+	NamespaceNodeLease string = "kube-node-lease"
```

Comment:
also, this doesn't really belong in meta/v1. it's fundamental to kube-apiserver and kubelet, but not to apimachinery (same really goes for kube-system and kube-public, but we can deal with those separately)

Reply:
`kube-node` would make sense to store node configmaps and leases. Thoughts?

Reply:
> would we only put leases in this namespace?

My understanding is just leases. This is what the KEP says:
>for each Node there will be a corresponding Lease object with Name equal to Node name in a newly created dedicated namespace (we considered using kube-system namespace but decided that it's already too overloaded). That namespace should be created automatically (similarly to "default" and "kube-system", probably by NodeController) and never be deleted (so that nodes don't require permission for it).

Which implies you just use it for node leases, because not prefixing the lease names means you can't cleanly add any other type of lease to that namespace in the future.

> needs to be communicated well in advance

My understanding is that we had already reserved the `kube-` prefix for namespaces? Maybe this isn't true? @thockin do you know?

> this doesn't really belong in meta/v1

That's fair, but then where should we put the constants for standard system namespaces? Multiple consumers need them, so they should be in a common place.

Reply:
> My understanding is that we had already reserved the `kube-` prefix for namespaces? Maybe this isn't true? @thockin do you know?

I've never seen that stated.

> where should we put the constants for standard system namespaces? Multiple consumers need them, so they should be in a common place.

For this, I'd probably start with a package visible both the kubelet and kube-apiserver. If it's truly a constant that cannot change across releases, then I wouldn't expect someone to take a difficult package dependency just for a string.

Reply:
Maybe just core/v1?

Reply:
looks like the others are already in `pkg/apis/core/types.go`... there or in core/v1/types.go is better than meta/v1

Reply:
I talked to @bgrant0607 about `kube-`, given that this is the convention we think we should just explicitly reserve it: https://github.com/kubernetes/community/pull/2379

Reply:
Why aren't the other ones in core/v1 in addition to core? Just because they're in meta/v1 instead (odd, as you said)?
Should put a constant for the node lease namespace in both core and core/v1?

Reply:
moved to both for now